platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,852 @@
+    g1Generator.isSameGroup(g2Generator); //false
+}
+```
+
+
+#### Swirlds Crypto Pairings Signature Library
+##### Overview
+This module provides cryptography primitives to create EC PublicKeys, EC PrivateKeys, and Signatures.
+
+##### Public API
+###### `SignatureSchema`
+**Link**: [SignatureSchema.java](signature-lib%2FSignatureSchema.java)
+
+**Description**: A pairings signature scheme can be implemented with different types of curves and group assignment configurations.
+For example, two different configurations might consist of a BLS_12_381 curve using G1 of the pairing to generate public key elements or G2 for the same purpose.
+###### `PairingPrivateKey`
+**Link**: [PairingPrivateKey.java](signature-lib%2FPairingPrivateKey.java)
+
+**Description**: A private key generated using the pairings API
+
+###### `PairingPublicKey`
+**Link**: [PairingPublicKey.java](signature-lib%2FPairingPublicKey.java)
+
+**Description**: A public key generated using the pairings API
+###### `PairingSignature`
+**Link**: [PairingSignature.java](signature-lib%2FPairingSignature.java)
+
+**Description**: A signature generated with the private key that can be verified with the public key
+
+###### `GroupAssignment`
+**Link**: [GroupAssignment.java](signature-lib%2FGroupAssignment.java)
+
+**Description**: An enum to clarify which group public keys and signatures are in, for a given SignatureSchema
+
+
+##### Implementation Note
+The serialization of the elements in this module adds a byte to represent the combination of Curve type and group assignment.
+
+##### Examples
+###### Generating a SignatureSchema
+```java
+SignatureSchema signatureSchema = SignatureSchema.from(Curve.ALT_BN128, GroupAssignament.G1_PUBLIC_KEYS);
+```
+###### Generating a Private Key
+```java
+   PairingPrivateKey privateKey = PairingPrivateKey.create(signatureSchema, new SecureRandom());
+   
+```
+###### Generating a Public Key
+```java
+   PairingPublicKey publicKey = PairingPublicKey.create(privateKey);
+   
+```
+###### Generating a Signature
+```java
+    byte[] message = new byte[]{};
+    PairingSignature signature = PairingSignature.sign(privateKey, message );
+   
+```
+###### Verifying a Signature
+```java
+static{
+    signature.verify(publicKey, message);
+}
+```
+##### Constraints
+This module will not depend on hedera-services artifacts, so it cannot include logging, metrics, configuration, or any other helper module from that repo.
+##### Dependencies
+swirlds-cryptography-pairings-API and runtime implementation
+
+
+#### Swirlds Crypto TSS Library
+##### Overview
+This library implements the Groth21 TSS-specific primitives.
+
+##### Constraints
+Once built, this library will depend on artifacts constructed in other repositories.
+This library accepts Integer.MAX_VALUE -1 participants.
+
+##### Public API:
+###### `TssService`
+**Description**: This class handles all tss specific operations.
```

Comment:
Will this service be stateless?

Reply:
Yes. No changing data structure is part of the initialization process either. We can hold a reference to the service or instantiate a new one each time it is needed.